include/ev3_robot.h

```diff
@@ -0,0 +1,95 @@
+/** \file ev3_robot.h
+ * \brief Naviugation funtions for a diffrential robot with the EV3 brick  for FLL Robots (POSE, Go , Gyro)
+ *
+ * ev3_robot.h contains declarations for the Robots movement functions.
+ *
+ * License:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/
+ *
+ * The Initial Developer of this code is Red Team FL.
+ * Portions created by Red Team FLL are Copyright (C) Red Team FLL.
+ * 
+ * 
+ * All Rights Reserved.
+ *
+ * ----------------------------------------------------------------------------
+ *
+ * \author Red Team FLL (redteamfll_at_gmail.com)
+ * \date 2020-08-15
+ * \version 0.0.1
+ */
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef robot_h
+#define robot_h
+
+#include "inputs/ev3_input.h"
+#include "sensors/ev3_sensors.h"
+#include "ev3.h"
+#include "math.h"
+#ifndef M_PI
+    #define M_PI		3.14159265358979323846	/* pi */
+#endif
+
+
+/**
+ * @brief Initialize pose at postion (x,y) (0,0) and angle (0).
+  *
+ *  */
+void PoseInit();
+
+
+/**
+ * @brief Initialize all robot variables, set POSE to (0,0,0),  * reset gyrosensor and reset tacho counters.
+ * @param port Port to which the sensor is connected
+ * @param mode How to measure the rate
+ * @param ColorLeft Port to which the left color sensor is connected 
+ * @param ColorRight Port to which the right color sensor is connected
+ * @param MotorLeft Port to wich the traction Left motor is connected
+ * @param MotorRight Port to wich the traction  right motor is connected
+ * @param Gyro Port to which the gyro sensor is connected 
+ * @param Debug If True print on the LCD screen the result of initialiation
+ * @return N/A.
+ */
+void RobotInit(int ColorLeft, int ColorRight, uint8_t MotorLeft, uint8_t MotorRight, int Gyro, bool Debug);
+
+/**
+ * @brief Initialize PID for navigation with girosensor.
```

Comment:
This function has been changed, now dont use parameters. All necessary robot data can be found in my_robot.h.
So, that's a definition file, do you think is better to name it like ev3_my_robot.h or my_robot.h, is correct?

Reply:
There is a problem with this approach, see https://github.com/c4ev3/EV3-API/pull/53#discussion_r473805719. That being said, the definition macros could be instead replaced with a struct that would be passed to the initialization function. The struct would be defined in `ev3_robot.h`:
```c
typedef struct {
    float WheelDiameter;
    float Width;
    int LeftMotorPort;
    int RightMotorPort;
    ...
} ROBOT_PARAMS;
```

In the user code, the structure would be used to define a new variable containing the parameters:
```c
ROBOT_PARAMS my_robot = {
    .WheelDiameter  = 56.0f, // mm
    .Width          = 100.0f, // mm
    .LeftMotorPort  = OUT_A,
    .RightMotorPort = OUT_B,
     ...
};
```

The initialization function then could accept a pointer to that structure:
```c
void RobotInit(ROBOT_PARAMS *params) {
    GlobalVars.WheelDiameter = params->WheelDiameter;
    ...
}
```

And the initialization function would be called like this:
```c
    // initialize with my robot definition
    RobotInit(&my_robot);
```

Reply:
GlobalVars is a variable of type ROBOT_PARAMS, correct? and in the rest of functions in robot c, we will use this global variable, right?
Well, we agreed with that, for the next commit, so this wil require an example for use, in which directory we can place it?

Reply:
> GlobalVars is a variable of type ROBOT_PARAMS, correct?

It can be, but it doesn't have to. Originally I thought of `ROBOT_PARAMS` as containing only unchanging data such as the wheel diameter. `GlobalVars` would contain the changing state (pose, ...) and then either a pointer to `ROBOT_PARAMS` or a copy of the parameters contained within.

> and in the rest of functions in robot c, we will use this global variable, right?

It is one of the possible solutions. Currently `MyRobot` behaves exactly like the `GlobalVars` would behave.

> Well, we agreed with that, for the next commit, so this wil require an example for use, in which directory we can place it?

`/examples/ev3_robot.c` should do.